dftimewolf/lib/args_validator.py

```diff
@@ -361,78 +403,99 @@ def Validate(self,
       dt = datetime.datetime.strptime(
           operand, validator_params['format_string'])
     except ValueError as exception:  # Date parsing failure
-      raise errors.RecipeArgsValidatorError(str(exception))
+      return False, str(exception)
 
-    if 'before' in validator_params and validator_params['before'] is not None:
-      self._ValidateOrder(
-          dt, validator_params['before'], validator_params["format_string"])
+    try:
+      if 'before' in validator_params and validator_params['before']:
+        if not self._ValidateOrder(
+            dt, validator_params['before'], validator_params["format_string"]):
+          return False, (f'{validator_params["before"]} is before {dt} but it '
+                         'should be the other way around')
+
+      if 'after' in validator_params and validator_params['after']:
+        if not self._ValidateOrder(
+            validator_params['after'], dt, validator_params["format_string"]):
+          return False, (f'{dt} is before {validator_params["after"]} but it '
+                         'should be the other way around')
+    except ValueError as exception:
+      raise errors.RecipeArgsValidatorError(
+          f'Error in order comparison: {str(exception)}')
```

Comment:
Shouldn't we be returning `'False', message` here?

Reply:
The error here is if there is a mismatch in the date format string, I think an exception is more appropriate.